src/main/generic/consensus/base/primitive/Hash.js

```diff
@@ -69,6 +77,7 @@ class Hash extends Serializable {
         switch (algorithm) {
             case Hash.Algorithm.BLAKE2B: return Hash.blake2b(arr);
             case Hash.Algorithm.SHA256: return Hash.sha256(arr);
+            case Hash.Algorithm.SHA512: return Hash.sha512(arr);
```

Comment:
This would result in a hard-fork (by adding a feature without backwards-compatibility), maybe postpone this to the next intended hard-fork?

Reply:
Good catch, I'll remove it from here.